src/lib/operators/projection.cpp

```diff
@@ -81,6 +82,40 @@ std::shared_ptr<const Table> Projection::_on_execute() {
         output_segments[column_id] = input_chunk->get_segment(pqp_column_expression->column_id);
         column_is_nullable[column_id] =
             column_is_nullable[column_id] || input_table.column_is_nullable(pqp_column_expression->column_id);
+      } else if (expression->type == ExpressionType::PQPColumn && !forward_columns) {
```

Comment:
yepp

Reply:
Negative, there are some cases where this is not true, e.g., for modified columns on a data table.

Reply:
Wow, good find! Did not think of this szenario. Maybe even worth a comment. ;)

Reply:
Well... "find" only in the sense that a bunch of tests flared up bright red. :D 
That's also the reason why I think that a comment is not really necessary, as it would not help in understanding the code.